presentation/src/main/java/org/sagebionetworks/research/presentation/speech/TextToSpeechService.kt

```diff
@@ -0,0 +1,437 @@
+        specialKeyMap["end"] = { duration -> duration.seconds }
+    }
+
+    // Queuing behavior is passed to all calls to TextToSpeech.speak(), default value is QUEUE_ADD
+    var queueingBehavior = TextToSpeech.QUEUE_ADD
+    // The duration of the sound played by playSound() and the vibration triggered by vibrate() in ms, default value
+    // is 500.
+    var defaultVibrateAndSoundDurationMillis: Long = 500
+    // The amount of time in seconds that speeches will still be spoken for if the service is initialized after
+    // the speeches exact time. (ie, if speechGracePeriod = 5 and the service is initialized 4 seconds after the step
+    // formally starts, instructions with the key "start" will still be spoken.
+    var speechGracePeriod: Long = 5
+    // The public getter for state should have type live data so client's don't modify it.
+    val state: LiveData<TextToSpeechState>
+        get() = _state
+    // The private state value is mutable so the service can update it's state as needed
+    private val _state = MutableLiveData<TextToSpeechState>()
+    private val bindCount = MutableLiveData<Int>()
+    private val cleanupMediator = MediatorLiveData<Pair<Int, TextToSpeechState>>()
+    private val cleanupObserver: Observer<Pair<Int, TextToSpeechState>>
+
+    init {
+        bindCount.postValue(0)
+        _state.postValue(TextToSpeechState(IDLE, null))
+        // We observe the state so that if every client unbinds and the service stops speaking, the service
+        // stops itself.
+        cleanupMediator.postValue(Pair(0, TextToSpeechState(IDLE, null)))
+        cleanupMediator.addSource(bindCount) { count ->
+            // these checks must be null safe to prevent the service from crashing when
+            cleanupMediator.postValue(
+                    Pair(count ?: 0,
+                            cleanupMediator.value?.second ?: TextToSpeechState(IDLE, null)))
+        }
+        cleanupMediator.addSource(_state) { s ->
+            cleanupMediator.postValue(Pair(cleanupMediator.value?.first ?: 0,
+                    s ?: TextToSpeechState(IDLE, null)))
+        }
+        cleanupObserver = Observer { pair ->
+            if (pair != null && pair.first == 0 && pair.second.speakingState != SPEAKING
+                    && pair.second.speakingState != QUEUED) {
+                stopSelf()
+            }
+        }
+
+        cleanupMediator.observeForever(cleanupObserver)
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        AndroidInjection.inject(this)
+    }
+
+    override fun onBind(intent: Intent?): IBinder {
+        bindCount.postValue((bindCount.value ?: 0) + 1)
+        if (LOGGER.isDebugEnabled) {
+            LOGGER.debug("onBind() called bindCount = $bindCount")
+        }
+
+        return serviceBinder
+    }
+
+    override fun onUnbind(intent: Intent?): Boolean {
+        bindCount.postValue(bindCount.value!! - 1)
```

Comment:
It's rare where you actually want to do the null-assertion operator, because you are disabling one of the best features of Kotlin.  Try this instead:

val bindCountValue = (bindCount.value ?: 1) - 1
bindCount.postValue(bindCountValue)